spring-cloud-dataflow-registry/src/main/java/org/springframework/cloud/dataflow/registry/service/DefaultAppRegistryService.java

```diff
@@ -413,14 +424,17 @@ private AppRegistration createAppRegistrations(Map<String, AppRegistration> regi
 				throw new IllegalArgumentException(e);
 			}
 		} else if (typeName.length == 3) {
+			if (extra.equals("metadata")) {
 				// metadata app uri
 				try {
 					ar.setMetadataUri(new URI(lineSplit[1]));
 					warnOnMalformedURI(lineSplit[0], ar.getMetadataUri());
 				} catch (Exception e) {
 					throw new IllegalArgumentException(e);
 				}
-
+			} else if (!"bootVersion".equals(extra)) {
```

Comment:
I don't think we need to check for `bootVersion` here, because the if block on lines 405-412, should return previous if, `"bootVersion".equals(extra)".   Thus extra should be something else.

Reply:
I will double-check. This was pulled over from the impl in `2.11.x` and simplified to just ignore the bootVersion line in the descriptor as the stream apps descriptors are already published w/ the bootVersion and to remove it would require stream apps release. I will see if we can simplify this further as suggested. Thanks.

Reply:
I believe this check is enforcing "the only 3 part keys in the descriptor file in here can be 'metadata' or 'bootVersion'". 

Keys in that file must be one of these:
- `<app-type>.<app-name>` 
- `<app-type>.<app-name>.(metadata|bootVersion)`

Without that check in here it would allow an entry such as `<app-type>.<app-name>.imposter`. I will verify this though before merging.

Reply:
Actually, you are right @cppwfs . The check does not nothing, in fact there is nothing that verifies the 3rd token is one of `metadata` or `bootVersion` because line 397 will fail on `getResourceVersion` if you put a key anything other than `metadata` in the extra field. 

E.g. this entry fails - but not in the if block in question (on line 397)...

```java
source.debezium=maven://org.springframework.cloud.stream.app:debezium-source-rabbit:5.0.0
source.debezium.fooVersion=3
source.debezium.metadata=maven://org.springframework.cloud.stream.app:debezium-source-rabbit:jar:metadata:5.0.0
```
Because `fooVersion` is indeed `!bootVersion` so it attempts to get version info that does not exist on the value of the entry (`3`).

Not going to make any changes for this behavior in this PR. Updated the code w/ your original request though. Thanks.